global-script.lua

```diff
@@ -2757,12 +2799,30 @@ function upCastPosSizRot(oPos,size,rot,dist,multi,tags)
 end
 ---- Block Square Section
 function setupPlayerArea(params)
-    --Sets position/color for the button, spawns it
-    params.obj.createButton({
+    -- Figure out what color we're supposed to be, or if playerswapping is even allowed.
+    local obj = params.obj
+    obj.createButton({
         label="Energy Cost: 0", click_function="nullFunc",
         position={0,2.24,-11.2}, rotation={0,180,0}, height=0, width=0,
         font_color={1,1,1}, font_size=500
     })
+
+    local color
+    for k, v in pairs(playerBlocks) do
+        if v.guid == obj.guid then
+            color = k
+            break
+        end
+    end
+    obj.setVar("playerColor", color)  -- May be nil
+    if color then
+        obj.createButton({
+            label="Move to here", click_function="onSwapButtonClicked", function_owner=Global,
```

Comment:
could you add a helpful tooltip that better explains what's going on? also wondering if there's a better phrase to put on the button. like maybe `Move [current color] to this seat" or something along those lines?

Reply:
How is this:

```lua
        obj.createButton({
            label="Move to here", click_function="onSwapButtonClicked", function_owner=Global,
            position={0,2.24,-7.2}, rotation={0,180,0}, height=800, width=4000,
            font_color={0,0,0}, font_size=500, 
            tooltip="Moves your current player color to be located here.  The color currently seated here will be moved to your current location.",
        })
```

I don't think it's possible to have buttons have different text based on which player is looking at them without a LOT of ugly trickery (like having six zones that are visible to only their respective players with customized texts.)

The buttons COULD say "Swap with Red", etc. though -- editButton() is a thing.

Reply:
guess i wasn't clear. was trying to say "Move to this seat" or "Move current color to this seat" was just saying that those words were optional not trying to save to plug in the actual color value.